release-process.rst

```diff
@@ -28,55 +34,135 @@ Roughly:
+exceptional cases where a SAPI presents significant maintenance or security
+concerns, an exception MAY be made. In all cases, the removal MUST go through
+the RFC process.
+
+All new user-facing features MUST be mentioned in the `UPGRADING
+<https://github.com/php/php-src/blob/master/UPGRADING>`_ document.
+
+All API and ABI breaks MUST be mentioned in the `UPGRADING.INTERNALS
+<https://github.com/php/php-src/blob/master/UPGRADING.INTERNALS>`_ document.
+
+Further reading:
+
+-  http://en.wikipedia.org/wiki/Application_programming_interface
+-  http://en.wikipedia.org/wiki/Application_binary_interface
 
 Major Version Number
 ====================
 
 -  x.y.z to x+1.0.0
 
-   -  Bug fixes
-   -  New features
-   -  Extensions support can be ended (moved to PECL)
-   -  Backward compatibility can be broken
-   -  API compatibility can be broken (internals and userland)
-   -  ABI can be broken (internals)
+   -  It SHALL include bugfixes and new features.
+   -  Extensions support MAY be ended (moved to PECL).
+   -  SAPI support MAY be ended (removed from the php-src repository).
+   -  Userland API backward compatibility MAY be broken.
+   -  Internal API backward compatibility MAY be broken.
+   -  ABI backward compatibility MAY be broken.
 
 Minor Version Number
 ====================
 
--  x.y.z to x.y+1.z
+-  x.y.z to x.y+1.0
 
-   -  Bugfixes
-   -  New features
-   -  Extensions support can be ended (moved to PECL)
-   -  Backward compatibility must be kept
-   -  API compatibility must be kept (userland)
-   -  ABI and API can be broken (internals)
-   -  Source compatibility should be kept if possible, while breakages are
-      allowed
+   -  It SHOULD include bugfixes and new features.
 
-Patch Version Number
-====================
+   -  Extensions support MAY be ended (moved to PECL).
 
--  x.y.z to x.y.z+1
+   -  SAPI support is RECOMMENDED to be kept.
 
-   -  Bug fixes and security patches only
-   -  Extensions support can't be removed (like move them to PECL)
-   -  Backward compatibility must be kept (internals and userland)
-   -  ABI and API compatibility must be kept (internals)
+   -  Syntax backward compatibility SHOULD be preserved - every PHP program that
+      compiles SHOULD continue to compile.
 
-It is critical to understand the consequences of breaking BC, APIs or ABIs (only
-internals related). It should not be done for the sake of doing it. RFCs
-explaining the reasoning behind a breakage and the consequences along with test
-cases and patch(es) should help.
+   -  Backward compatibility breaks in minor versions MUST NOT result in silent
+      behavioral differences. Instead any breaking change MUST be "obvious" when
+      executing the program. It means it SHOULD either throw exception or
+      trigger error.
 
-See the following links for explanation about API and ABI:
+   -  Userland API backward compatibility breaks SHOULD be limited to
+      extensions, or the API of functions within an extension.
```

Comment:
I'd like to see *something* about “BC breaks must be simple to work-around in a cross-version way”, e.g. what I mentioned in https://externals.io/message/127328#127335.

Reply:
@TimWolla What do you think about https://github.com/php/policies/pull/19/commits/a6cd5320fa1cf5c8bee404ea8e2e0dc3942aee3b ? I decided to use SHOULD as it should be more recommendation and a hint for voters to decide about it rather than being fully restrictive. It's also more inline with what we have now.